examples/v1beta1/pipelineruns/alpha/pipeline-emitting-default-results.yaml

```diff
@@ -0,0 +1,72 @@
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: pipelinerun-emit-default-results
+spec:
+  pipelineSpec:
+    tasks:
+      - name: task1
+        taskSpec:
+          results:
+            - name: array-results
+              type: array
+              description: The array results
+              default:
+              - "1"
+              - "2"
+              - "3"
+          steps:
+            - name: write-array
+              image: bash:latest
+              script: |
+                #!/usr/bin/env bash
+                echo -n "this script won't emit array result"
```

Comment:
I wanted to highlight the issue https://github.com/tektoncd/pipeline/issues/3497 which probably seems to go against this example?

Reply:
The discussion in that issue hasn't come to a conclusion so this might be ok for now. I wonder if your examples could genuinely error out and the steps have an `onError` --> `continue` and then have the default result use?